internal/db/db.go

```diff
@@ -80,6 +81,15 @@ type db struct {
 	// The peer ID and network address information for the current node
 	// if network is enabled. The `atomic.Value` should hold a `peer.AddrInfo` struct.
 	peerInfo atomic.Value
+
+	// To be able to close the context passed to NewDB on DB close,
+	// we need to keep a reference to the cancel function. Otherwise,
+	// some goroutines might leak.
+	ctxCancel context.CancelFunc
+
+	retryIntervals []time.Duration
+	retryChan      chan event.ReplicatorFailure
+	retryDone      chan retryStatus
```

Comment:
suggestion: `db` is quite a busy type, it might be worth rolling these 3 props up into a new type.

Reply:
✅